src/ekat/util/ekat_factory.hpp

```diff
@@ -1,13 +1,13 @@
 #ifndef EKAT_FACTORY_HPP
 #define EKAT_FACTORY_HPP
 
+#include "ekat/ekat_assert.hpp"
```

Comment:
Another question for my own curiosity: did you have to move this above the other includes to satisfy the compiler? It's more conventional for "home-made" headers to go below standard ones.

Reply:
I usually follow the approach of going from local to global (which I thought was more common), to avoid the scenario where the code compiles just because you happen to include things in the right order. If, say, `<string>` is needed in `ekat_type_traits.hpp`, I want it to be included there. If I put std headers first, that lack of include directive in `ekat_tyep_traits.hpp` might go undetected.

Reply:
That makes sense.